source/extensions/filters/listener/proxy_protocol/proxy_protocol.cc

```diff
@@ -87,8 +146,27 @@ bool Config::isPassThroughTlvTypeNeeded(uint8_t tlv_type) const {
 
 size_t Config::numberOfNeededTlvTypes() const { return tlv_types_.size(); }
 
-bool Config::allowRequestsWithoutProxyProtocol() const {
-  return allow_requests_without_proxy_protocol_;
+bool Config::isVersionAllowed(ProxyProtocolVersion version) const {
+  switch (version) {
+  case Unknown:
+    return allow_requests_without_proxy_protocol_;
+  case ProxyProtocolVersion::V1:
+    return allow_v1_;
+  case ProxyProtocolVersion::V2:
+    return allow_v2_;
+  }
+}
+
+VersionedProxyProtocolStats& Config::versionToStatsStruct(ProxyProtocolVersion version) {
+  switch (version) {
+  case Unknown:
+    return stats_.unknown_;
+  case ProxyProtocolVersion::V1:
+    return stats_.v1_;
+  case ProxyProtocolVersion::V2:
+    return stats_.v2_;
+  }
+  return stats_.unknown_; // Should never happen.
```

Comment:
Make the type an `enum class` and then you can delete this line; the compiler will error on an unhandled case in the switch.

Reply:
Done, good idea.

Reply:
Fails windows build :( Adding back the default paths

```
INFO: From Compiling src/cpp/common/alarm.cc:
source\extensions\filters\listener\proxy_protocol\proxy_protocol.cc(169) : error C2220: the following warning is treated as an error
source\extensions\filters\listener\proxy_protocol\proxy_protocol.cc(158) : error C2220: the following warning is treated as an error
source\extensions\filters\listener\proxy_protocol\proxy_protocol.cc(158) : warning C4715: 'Envoy::Extensions::ListenerFilters::ProxyProtocol::Config::isVersionAllowed': not all control paths return a value
source\extensions\filters\listener\proxy_protocol\proxy_protocol.cc(169) : warning C4715: 'Envoy::Extensions::ListenerFilters::ProxyProtocol::Config::versionToStatsStruct': not all control paths return a value
ERROR: C:/source/source/extensions/filters/listener/proxy_protocol/BUILD:14:17: Compiling source/extensions/filters/listener/proxy_protocol/proxy_protocol.cc failed: (Exit 2): cl.exe failed: error executing command (from target //source/extensions/filters/listener/proxy_protocol:proxy_protocol_lib) 
```